apis/extensions/v1alpha1/addon_types.go

```diff
@@ -319,6 +323,18 @@ type CliPlugin struct {
 	Description string `json:"description,omitempty"`
 }
 
+type AddonDependency struct {
+	// The name of the dependent addon.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// All matching versions of the dependent addon. If empty, defaults to the same version as the current addon.
+	//
+	// +optional
+	Version []string `json:"version"`
```

Comment:
I hope the version specified in this array could express range.

e.g. like the python or ruby, ">=0.8.3"